src/fiatExchanges/SelectProvider.tsx

```diff
@@ -158,39 +166,89 @@ export default function SelectProviderScreen({ route, navigation }: Props) {
+
+  const switchCurrencyOnPress = () =>
+    navigate(Screens.FiatExchangeCurrency, {
+      flow:
+        route.params.flow === CICOFlow.CashIn ? FiatExchangeFlow.CashIn : FiatExchangeFlow.CashOut,
+    })
+
   const exchanges = asyncExchanges.result ?? []
+  const legacyMobileMoneyProviders = asyncProviders.result?.legacyMobileMoneyProviders
+
+  const coinbasePayVisible =
+    coinbaseProvider &&
+    !coinbaseProvider.restricted &&
+    coinbasePayEnabled &&
+    appId &&
+    digitalAsset === CiCoCurrency.CELO
+
+  const anyProviders =
+    normalizedQuotes.length ||
+    coinbasePayVisible ||
+    exchanges.length ||
+    legacyMobileMoneyProviders?.length
 
   return (
-    <ScrollView>
-      <PaymentMethodSection
-        normalizedQuotes={normalizedQuotes}
-        paymentMethod={PaymentMethod.Card}
-        setNoPaymentMethods={setNoPaymentMethods}
-        flow={flow}
-      />
-      <PaymentMethodSection
-        normalizedQuotes={normalizedQuotes}
-        paymentMethod={PaymentMethod.Bank}
-        setNoPaymentMethods={setNoPaymentMethods}
-        flow={flow}
-      />
-      <LegacyMobileMoneySection
-        providers={asyncProviders.result?.legacyMobileMoneyProviders || []}
-        digitalAsset={digitalAsset}
-        flow={flow}
-      />
-      <CoinbasePaymentSection
-        digitalAsset={digitalAsset}
-        cryptoAmount={route.params.amount.crypto}
-        coinbaseProvider={coinbaseProvider}
-      />
-      <ExchangesSection
-        exchanges={exchanges}
-        selectedCurrency={route.params.selectedCrypto}
-        flow={flow}
-      />
-      <LimitedPaymentMethods visible={noPaymentMethods} flow={flow} />
-    </ScrollView>
+    <SafeAreaView style={styles.container}>
+      {anyProviders ? (
+        <ScrollView>
+          <PaymentMethodSection
+            normalizedQuotes={normalizedQuotes}
+            paymentMethod={PaymentMethod.Card}
+            setNoPaymentMethods={setNoPaymentMethods}
+            flow={flow}
+          />
+          <PaymentMethodSection
+            normalizedQuotes={normalizedQuotes}
+            paymentMethod={PaymentMethod.Bank}
+            setNoPaymentMethods={setNoPaymentMethods}
+            flow={flow}
+          />
+          <LegacyMobileMoneySection
+            providers={legacyMobileMoneyProviders || []}
+            digitalAsset={digitalAsset}
+            flow={flow}
+          />
+          {coinbaseProvider && coinbasePayVisible ? (
+            <CoinbasePaymentSection
+              cryptoAmount={route.params.amount.crypto}
+              coinbaseProvider={coinbaseProvider}
+              appId={appId}
+            />
+          ) : null}
+          <ExchangesSection
+            exchanges={exchanges}
+            selectedCurrency={route.params.selectedCrypto}
+            flow={flow}
+          />
+          <LimitedPaymentMethods visible={noPaymentMethods} flow={flow} />
+        </ScrollView>
+      ) : (
```

Comment:
nit: I'm not a fan of this very large ternary operator. I think it'd be more clear in the code if you had a large if statement returning the change currency UI. Similar to the Loading screen logic

```
if(!anyProviders) {
// return switch currency UI
}
```